jgrapht-core/src/main/java/org/jgrapht/generate/PlantedPartitionGraphGenerator.java

```diff
@@ -0,0 +1,258 @@
+     */
+    public PlantedPartitionGraphGenerator(int l, int k, double p, double q, long seed)
+    {
+        this(l, k, p, q, new Random(seed), DEFAULT_ALLOW_SELFLOOPS);
+    }
+
+    /**
+     * Construct a new PlantedPartitionGraphGenerator.
+     *
+     * @param l number of groups
+     * @param k number of nodes in each group
+     * @param p probability of connecting vertices within a group
+     * @param q probability of connecting vertices between groups
+     * @param seed seed for the random number generator
+     * @param selfLoops true if the graph allows self loops
+     * @throws IllegalArgumentException if number of groups is negative
+     * @throws IllegalArgumentException if number of nodes in each group is negative
+     * @throws IllegalArgumentException if p is not in [0,1]
+     * @throws IllegalArgumentException if q is not in [0,1]
+     */
+    public PlantedPartitionGraphGenerator(int l, int k, double p, double q, long seed, boolean selfLoops)
+    {
+        this(l, k, p, q, new Random(seed), DEFAULT_ALLOW_SELFLOOPS);
+    }
+
+    /**
+     * Construct a new PlantedPartitionGraphGenerator.
+     *
+     * @param l number of groups
+     * @param k number of nodes in each group
+     * @param p probability of connecting vertices within a group
+     * @param q probability of connecting vertices between groups
+     * @param rng random number generator
+     * @param selfLoops true if the graph allows self loops
+     * @throws IllegalArgumentException if number of groups is negative
+     * @throws IllegalArgumentException if number of nodes in each group is negative
+     * @throws IllegalArgumentException if p is not in [0,1]
+     * @throws IllegalArgumentException if q is not in [0,1]
+     */
+    public PlantedPartitionGraphGenerator(int l, int k, double p, double q, Random rng, boolean selfLoops)
+    {
+        if (l < 0) {
+            throw new IllegalArgumentException("number of groups must be non-negative");
+        }
+        if (k < 0) {
+            throw new IllegalArgumentException("number of nodes in each group must be non-negative");
+        }
+        if (p < 0 || p > 1) {
+            throw new IllegalArgumentException("invalid probability p");
+        }
+        if (q < 0 || q > 1) {
+            throw new IllegalArgumentException("invalid probability q");
+        }
+        this.l = l;
+        this.k = k;
+        this.p = p;
+        this.q = q;
+        this.rng = rng;
+        this.selfLoops = selfLoops;
+    }
+
+    /**
+     * Generate an $l$-planted partition graph.
+     *
+     * @param target target graph
+     * @param vertexFactory vertex factory
+     * @param resultMap result map
+     * @throws IllegalArgumentException if target is directed
+     * @throws IllegalArgumentException if self loops are requested but target does not allow them
+     */
+    @Override
+    public void generateGraph(Graph<V, E> target, VertexFactory<V> vertexFactory, Map<String, V> resultMap)
```

Comment:
Would it be useful to communicate back to the user which vertices belong to which group? If so, consider using the resultMap for that. I realize that a `Map<String,V>` is not an ideal map to do this. If this information is not useful, ignore this comment.

Reply:
Yep, this is something I wanted to add too.

I don't know what is the best way to do it, but to have such information we need some mapping between nodes and community nodes belong to.

Especially considering a potential support for community detection algorithm, it could be the case that this mapping could be introduced by an ad-hoc class (e.g. CommunityStructure or Clustering).

Reply:
If you think it is useful, consider adding a List<Set<V>> getCommunities() method? (Use LinkedHashSets).